server/src/test/java/io/druid/client/client/BatchServerInventoryViewTest.java

```diff
@@ -244,14 +256,14 @@ public void testRunWithFilter() throws Exception
     Set<DataSegment> segments = Sets.newHashSet(server.getSegments().values());
 
     Assert.assertEquals(testSegments, segments);
-
+    int prevUpdateCount = inventoryUpdateCounter.get();
     // segment outside the range of default filter
     DataSegment segment1 = makeSegment(101);
     segmentAnnouncer.announceSegment(segment1);
     testSegments.add(segment1);
 
-    exception.expect(ISE.class);
-    waitForSync(filteredBatchServerInventoryView, testSegments);
+    waitForUpdateEvents(prevUpdateCount+1);
```

Comment:
I'm missing something here. Shouldn't this loop forever since `inventoryUpdateCounter` should be 100 before the announcement and 101 after the announcement?